Login Error

When I execute `1d login` I get the following error.

`[ERROR] Error while refreshing access token Error: The clientSecret parameter is required.
    at Object.validateStringParameter (/usr/lib/node_modules/@adobe/helix-onedrive-cli/node_modules/adal-node/lib/argument.js:37:13)
    at AuthenticationContext.acquireTokenWithClientCredentials (/usr/lib/node_modules/@adobe/helix-onedrive-cli/node_modules/adal-node/lib/authentication-context.js:282:14)
    at /usr/lib/node_modules/@adobe/helix-onedrive-cli/node_modules/@adobe/helix-onedrive-support/src/OneDrive.js:170:17
    at new Promise (<anonymous>)
    at OneDrive.getAccessToken (/usr/lib/node_modules/@adobe/helix-onedrive-cli/node_modules/@adobe/helix-onedrive-support/src/OneDrive.js:148:12)
    at Object.login (/usr/lib/node_modules/@adobe/helix-onedrive-cli/src/onedrive.js:47:34)
The clientSecret parameter is required.`

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
   "homepage": "https://github.com/adobe/helix-onedrive-cli#readme",
   "dependencies": {
     "@adobe/helix-log": "4.5.2",
-    "@adobe/helix-onedrive-support": "3.1.1",
+    "@adobe/helix-onedrive-support": "3.1.2",
     "chalk": "4.1.0",
     "cookie-parser": "1.4.5",
     "dotenv": "8.2.0",

--- a/src/client.js
+++ b/src/client.js
@@ -16,7 +16,8 @@ const { debug, info, SimpleInterface } = require('@adobe/helix-log');
 
 const STATE_FILE = '.hlx-1d.json';
 const AUTH_FILE = '.auth.json';
-const DEFAULT_CLIENT_ID = 'f4c79ff7-bbd2-4b36-822e-a89eb6de4578';
+// default client id is from microsoft's adal client
+const DEFAULT_CLIENT_ID = '04b07795-8ddb-461a-bbee-02f9e1bf7b46';
 
 let state = {};
 

--- a/src/onedrive.js
+++ b/src/onedrive.js
@@ -44,10 +44,12 @@ async function prompt(msg, hide = false) {
 
 async function login(username, password) {
   const od = await getOneDriveClient();
-  const tokenResponse = await od.getAccessToken();
-  if (tokenResponse.refreshToken) {
-    info('already logged in.');
-    return;
+  if (od.refreshToken) {
+    const tokenResponse = await od.getAccessToken();
+    if (tokenResponse.refreshToken) {
+      info('already logged in.');
+      return;
+    }
   }
   if (username !== undefined) {
     if (!username) {
@@ -64,15 +66,14 @@ async function login(username, password) {
     }
     od.username = username;
     od.password = password;
-    await od.getAccessToken();
-    const result = await od.me();
-    info(chalk`Logged in as: {yellow ${result.displayName}} {grey (${result.mail})}`);
-    return;
+  } else {
+    await od.login(async (code) => {
+      await openBrowser(code.verificationUrl);
+    });
   }
 
-  await od.login(async (code) => {
-    await openBrowser(code.verificationUrl);
-  });
+  const result = await od.me();
+  info(chalk`Logged in as: {yellow ${result.displayName}} {grey (${result.mail})}`);
 }
 
 async function logout() {

